### PR TITLE
Documentation to readiness framework

### PR DESCRIPTION
**Pull request by sathyanarays**
### What this PR does / why we need it
This PR adds more documentation to the readiness framework. This PR includes two different documents:
- How to deploy the local changes to a cluster (for dev purpose)
- Basic usage of the framework with a simple, working example

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

File: docs/runtime-core/readiness-framework/deploy-local-changes.md
[the remark this answers is not in the document]
Comment (sathyanarays) [reply docs/runtime-core/readiness-framework/deploy-local-changes.md:1]: Yes, we will be doing that going forward and replace all the instructions here with single make command.

File: docs/runtime-core/readiness-framework/deploy-local-changes.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+```
+make manifests
+```
+
+## Step 3 - Sync the generated manifests
+
+This step is required only if there is a change in the readiness APIs. Run the following commands to sync the newly generated CRDs to the packages directory.
+
+```
+make manifests
```
[the remark this answers is not in the document]
Comment (sathyanarays) [reply docs/runtime-core/readiness-framework/deploy-local-changes.md]: Done

File: docs/runtime-core/readiness-framework/deploy-local-changes.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```
+
+## Step 4 - Build docker image
+
+Run the following command from the `tanzu-framework` directory.
+
+```
+make docker-build-all
```
[the remark this answers is not in the document]
Comment (sathyanarays) [reply docs/runtime-core/readiness-framework/deploy-local-changes.md]: Done

File: docs/runtime-core/readiness-framework/guide-with-examples.md [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+| dev                    | package-management                                            |
+| canary                 | package-management, secret-management                         |
+| prod                   | package-management, secret-management, certificate-management |
+
+
+The manifests for the above readiness definitions are as follows
+
+```
+apiVersion: core.tanzu.vmware.com/v1alpha2
+kind: Readiness # CapabilityClass
```
[the remark this answers is not in the document]
Comment (sathyanarays) [reply docs/runtime-core/readiness-framework/guide-with-examples.md]: Removed!

**Review by avi-08 — APPROVED** [review]
LGTM, please fix the doc lint issues.

**Review by yharish991 — APPROVED** [review]

**Review by rajathagasthya — APPROVED** [review]

**Pull request merged**